src/main/scala/org/ergoplatform/mining/ErgoMiner.scala

```diff
@@ -412,6 +457,17 @@ class ErgoMiner(ergoSettings: ErgoSettings,
     readersHolderRef ! GetReaders
   }
 
+  // Start internal miner's threads. Called once per block candidate.
+  private def startInternalMiner(candidateBlock: CandidateBlock): Unit = {
+    secretKeyOpt match {
+      case Some(sk) =>
+        miningThreads += ErgoMiningThread(ergoSettings, self, candidateBlock, sk.w, timeProvider)(context)
```

Comment:
this will create a new actor on each block candidate. And those actors only destroyed in postStop, which means this array is growing on every block candidate. Looks suspicious.